mail/api.py

```diff
@@ -285,6 +285,76 @@ def send_course_team_email(  # pylint: disable=too-many-arguments
         )
         return response
 
+    @staticmethod
+    def add_email_to_unsub_list(url, email):
+        """
+        It adds user email to mailgun unsub list.
+
+        Args:
+            url (str): mailgun api url:
+            email (str): user email
+        """
+        try:
+            response = requests.post(
+                url,
+                auth=('api', settings.MAILGUN_KEY),
+                data={
+                    'address': email,
+                    'tag': '*'
+                }
+            )
+        except requests.exceptions.RequestException:
+            log.exception(
+                "Unable to add email: %s to mailgun unsubscribes list.", email
+            )
+            return
+        if response.status_code == status.HTTP_200_OK:
+            log.debug(
+                "Added user's email: %s to mailgun unsubscribes list. Message received: %s",
+                email,
+                response.json()
+            )
+
+    @staticmethod
+    def remove_email_from_unsub_list(url, email):
+        """
+        It removes user email from mailgun unsub list.
+
+        Args:
+            url (str): mailgun api url:
+            email (str): user email
+        """
+        try:
+            response = requests.delete(url, auth=('api', settings.MAILGUN_KEY))
+        except requests.exceptions.RequestException:
+            log.exception(
+                "Unable to remove email: %s from mailgun unsubscribes list.", email
+            )
+            return
+        if response.status_code == status.HTTP_200_OK:
+            log.debug(
+                "Removed user's email: %s from mailgun unsubscribes list. Message received: %s",
+                email,
+                response.json()
+            )
+
+    @staticmethod
+    def toggle_mailing_list_subscription(email_optin, email):
```

Comment:
Sorry, I should have noticed this during the last review. I think this method should be removed, and `ProfileViewset.perform_update` can inspect the `email_optin` value and decide to call `add_email_to_unsub_list` or `remove_email_to_unsub_list`. The `add`/`remove` methods can declare the url on their own instead of passing it in, so both of them would take `email` as their only argument